### PR TITLE
Add icon name to descriptor

### PR DESCRIPTION
**Pull request by halkeye**
Merged in icon-tag-shim-test and then extended it to include icon class name in the descriptor.

I read over the contributing guidelines. I can't see anything not setup right.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -106,6 +106,11 @@
       <version>1.3.9</version>
     </dependency>
     <!-- plugin dependencies -->
+    <dependency>
+      <groupId>org.jenkins-ci.plugins.icon-shim</groupId>
+      <artifactId>icon-shim</artifactId>
+      <version>1.0.3</version>
```
Comment (tfennelly) [inline pom.xml]: Seems like `2.0.1` is the latest release. Might be good to try with that.

**Comment by tfennelly** [comment]
@halkeye LGTM.

Thanks for fixing this up. Not sure why we didn't merge the original commit. I see no reason not to do it.

**Comment by halkeye** [comment]
@tfennelly Updating to 2.0.1 seems to break things. 
I changed the pom file, ran hpi:run and have no images.

I'm thinking it has something todo with https://github.com/jenkinsci/icon-shim-plugin/compare/icon-shim-pom-1.0.3...icon-shim-pom-2.0.1#diff-c0aec591762b9fcb07f322e6bb962df5L95

Is there something I need to do to get the icons into css now?

**Comment by halkeye** [comment]
@tfennelly infact the readme on still says to use 1.0.3 (https://github.com/jenkinsci/icon-shim-plugin)

**Comment by halkeye** [comment]
Yea, looks like 2.0 is a whole different design, but I could update to latest 1.0, but the readme for the repo/module says 1.0.3 so was kinda sticking with it.

Pinging @stephenc as he looks like the current maintainer for a review

**Comment by halkeye** [comment]
Okay, updated for 2.0.2 as @tfennelly updated it to make the shim work properly again.

**Comment by oleg-nenashev** [comment]
LGTM :+1:

File: src/main/java/com/cloudbees/plugins/credentials/PluginImpl.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2004-2009, Sun Microsystems, Inc., Kohsuke Kawaguchi
```
Comment (KostyaSha) [inline src/main/java/com/cloudbees/plugins/credentials/PluginImpl.java]: copy-paste?

Comment (halkeye) [inline src/main/java/com/cloudbees/plugins/credentials/PluginImpl.java]: you'll have to ask @tfennelly thats his patch

**Pull request merged**